src/librustc_metadata/rmeta/encoder.rs

```diff
@@ -1578,6 +1597,10 @@ impl EncodeContext<'tcx> {
         }
     }
 
+    fn encode_ident_span(&mut self, def_id: DefId, ident_span: Span) {
```

Comment:
```suggestion
    fn encode_ident_span(&mut self, def_id: DefId, ident: Ident) {
```
Could you pass a full `Ident` to here, so it would be easier to make an experiment for encoding/decoding full `Ident`s instead of combining them from `Symbol` and `Span` during decoding.